File: go/src/sort/bubbleSort.go
package sort

func bubbleSort(arr []int) {
	l := len(arr)

	if l < 2 {
		return
	}

	var swapped bool
	for i := 0; i < l-1; i++ {
		swapped = false
		for j := l - 1; j > 0; j-- {
			if arr[j] < arr[j-1] {
				arr[j], arr[j-1] = arr[j-1], arr[j]
				swapped = true
			}
		}
		if !swapped {
			break
		}
	}
}

File: archive/0_Other/Python/bubbleSort.py
def bubbleSort(arr):
    ln = len(arr)

    for i in range(ln):
        for j in range(ln - i - 1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]


arr = [64, 34, 25, 12, 22, 11, 90]

bubbleSort(arr)

print("排序后的数组:")
for i in range(len(arr)):
    print("%d" % arr[i]),
